src/core/qgsproject.h

```diff
@@ -575,6 +576,54 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
      */
     QMap<QString, QgsMapLayer *> mapLayers() const;
 
+    /**
+     * Unzip a project
+     * \param filename The zip file to unzip
+     * \returns true if unzip is well performed, false otherwise
+     * \since QGIS 3.0
+     */
+    bool unzip( const QString &filename );
+
+    /**
+     * Unzip a project with the current zip filename
+     * \returns true if unzip is well performed, false otherwise
+     * \since QGIS 3.0
+     */
+    bool unzip();
+
+    /**
+     * Zip the project
+     * \param filename The zip filename
+     * \returns true if zip is well performed, false otherwise
+     * \since QGIS 3.0
+     */
+    bool zip( const QString &filename );
```

Comment:
Actually, it was my first approach. 

If @nyalldawson is agree with that too, I can go on this way?